frontend/test/validate-e2e-test-files.js

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env node
+const { execSync } = require("child_process");
+
+const path = require("path");
+const glob = require("glob");
+const chalk = require("chalk");
+
+const e2eFileExtension = ".cy.spec.js";
+const e2eHome = "frontend/test/metabase/scenarios/";
+
+init();
+
+function validateStagedFiles() {
+  const stagedFiles = execSync("git diff HEAD --name-only --diff-filter=d", {
+    encoding: "utf8",
+  });
+  const relevantStagedFiles = stagedFiles.includes(e2eHome);
+  if (!relevantStagedFiles) {
+    return;
+  } else {
+    const invalidFileNames = stagedFiles
+      .split("\n")
+      .filter(fullPath => {
+        const dirName = path.dirname(fullPath);
+        const excludedPaths =
+          dirName.endsWith("/helpers") || dirName.endsWith("/shared");
+        return dirName.startsWith(e2eHome) && !excludedPaths;
+      })
+      .filter(fullPath => {
+        return !path.basename(fullPath).endsWith(e2eFileExtension);
+      });
+
+    printFeedback(invalidFileNames);
+  }
+}
+
+function validateAllFiles() {
+  // Will match all files in the scenarios dir, except the helpers
+  const PATTERN =
+    "frontend/test/metabase/scenarios/*/{*.js,!(helpers|shared)/*.js}";
+  const invalidFileNames = glob.sync(PATTERN).filter(fullPath => {
```

Comment:
I wish these 2 functions (`validateStagedFiles`, and `validateAllFiles`) look a bit more similar, the different part is where they get their own list of files, but the way it validates the files should be the same.

I wish they read something like this instead  (pseudo code)
```
function validateStagedFiles() {
  filesToCheck = getStagedFiles()
  invalidFileNames = getInvalidFileNames(filesToCheck)
  printFeedback(invalidFileNames)
```
and
```
function validateAllFiles() {
  filesToCheck = getAllFiles()
  invalidFileNames = getInvalidFileNames(filesToCheck)
  printFeedback(invalidFileNames)
```